src/UnityExtension/Assets/Editor/GitHub.Unity/UI/GitPathView.cs

```diff
@@ -0,0 +1,286 @@
+                        {
+                            GUI.FocusControl(null);
+                            isBusy = true;
+
+                            ValidateAndSetGitInstallPath(newGitExec);
+                        }
+                    }
+                    EditorGUI.EndDisabledGroup();
+
+                    //Find button - for attempting to locate a new install
+                    if (GUILayout.Button(GitInstallFindButton, GUILayout.ExpandWidth(false)))
+                    {
+                        GUI.FocusControl(null);
+                        isBusy = true;
+
+                        newGitExec = gitExec;
+                        CheckEnteredGitPath();
+
+                        new ProcessTask<NPath>(Manager.CancellationToken, new FirstLineIsPathOutputProcessor())
+                            .Configure(Manager.ProcessManager, Environment.IsWindows ? "where" : "which", "git")
+                            .FinallyInUI((success, ex, path) =>
+                            {
+                                if (success)
+                                {
+                                    Logger.Trace("FindGit Path:{0}", path);
+                                }
+                                else
+                                {
+                                    if (ex != null)
+                                    {
+                                        Logger.Error(ex, "FindGit Error Path:{0}", path);
+                                    }
+                                    else
+                                    {
+                                        Logger.Error("FindGit Failed Path:{0}", path);
+                                    }
+                                }
+
+                                if (success)
+                                {
+                                    newGitExec = path;
+                                    CheckEnteredGitPath();
+                                }
+
+                                isBusy = false;
+                            }).Start();
+                    }
+                }
+                GUILayout.EndHorizontal();
+
+                if (gitFileErrorMessage != null)
+                {
+                    GUILayout.BeginHorizontal();
+                    {
+                        GUILayout.Label(gitFileErrorMessage, Styles.ErrorLabel);
+                    }
+                    GUILayout.EndHorizontal();
+                }
+
+                if (gitVersionErrorMessage != null)
+                {
+                    GUILayout.BeginHorizontal();
+                    {
+                        GUILayout.Label(gitVersionErrorMessage, Styles.ErrorLabel);
+                    }
+                    GUILayout.EndHorizontal();
+                }
+            }
+            EditorGUI.EndDisabledGroup();
+        }
+
+        private void MaybeUpdateData()
+        {
+            if (gitExecHasChanged)
+            {
+                if (Environment != null)
```

Comment:
This is never not null since it comes directly from the cache and is automatically created on first access

Reply:
👍